test/e2e/framework/test_context.go

```diff
@@ -203,9 +205,11 @@ func RegisterCommonFlags() {
 	flag.StringVar(&TestContext.FeatureGates, "feature-gates", "", "A set of key=value pairs that describe feature gates for alpha/experimental features.")
 	flag.StringVar(&TestContext.Viper, "viper-config", "e2e", "The name of the viper config i.e. 'e2e' will read values from 'e2e.json' locally.  All e2e parameters are meant to be configurable by viper.")
 	flag.StringVar(&TestContext.ContainerRuntime, "container-runtime", "docker", "The container runtime of cluster VM instances (docker/rkt/remote).")
-	flag.StringVar(&TestContext.ContainerRuntimeEndpoint, "container-runtime-endpoint", "", "The container runtime endpoint of cluster VM instances.")
+	flag.StringVar(&TestContext.ContainerRuntimeEndpoint, "container-runtime-endpoint", "unix:///var/run/dockershim.sock", "The container runtime endpoint of cluster VM instances.")
```

Comment:
Why not relying on kubelet's defaults? Does this mean if we change the dockershim socket in kubelet, we'd have to change the test as well?

Also, the relationship between `--container-runtime` and the new flags may be hard for people (w/o prior knowledge of kubelet internals) to understand.

Reply:
The problem is that when connecting the container runtime in the test, we need to know where to connect anyway.

Unless we add the default in `kubelet/remote`, we have to set the default in the test somewhere.

Reply:
Do we also need the defaults set in `hack/make-rules/test-e2e-node.sh`?

Reply:
Another option is to skip setting endpoint in `hack/make-rules/test-e2e-node.sh` if it's empty. I can do that.

Reply:
Done.